### ISSUE TITLE
KSQL should support at-sign in columns

### ISSUE THREAD

**Issue by rmoff** [confluentinc/ksql#2348]
Data from sources including Elastic (e.g. Logstash, beats) can use the `@` at sign in column names, e.g.: 

```
{"@timestamp": "2019-01-14T14:34:47.617Z","message": "20190114T063447-0800,dm-2,SSD2T-backarea,1.99,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00","node": "hostnameA"}
```

KSQL doesn't handle this well. 

Load demo data: 

```
echo '{"@timestamp": "2019-01-14T14:34:47.617Z","message": "20190114T063447-0800,dm-2,SSD2T-backarea,1.99,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00","node": "hostnameA"}' | kafkacat -b kafka:29092 -t demo -P
```

PRINT: 

```
ksql> print 'demo' from beginning;
Format:JSON
{"ROWTIME":1547535783153,"ROWKEY":"null","@timestamp":"2019-01-14T14:34:47.617Z","message":"20190114T063447-0800,dm-2,SSD2T-backarea,1.99,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00","node":"hostnameA"}
```

`CREATE STREAM` with no quote - doesn't work

```
ksql> CREATE STREAM demo_as_json (@timestamp varchar, message varchar, node varchar) with (kafka_topic='demo', value_format='json');
line 1:29: extraneous input '@' expecting {'NO', 'INTEGER', 'DATE', 'TIME', 'TIMESTAMP', 'INTERVAL', 'YEAR', 'MONTH', 'DAY', 'HOUR', 'MINUTE', 'SECOND', 'ZONE', 'PARTITION', 'STRUCT', 'REPLACE', 'EXPLAIN', 'ANALYZE', 'FORMAT', 'TYPE', 'TEXT', 'SHOW', 'TABLES', 'SCHEMAS', 'COLUMNS', 'COLUMN', 'PARTITIONS', 'FUNCTIONS', 'FUNCTION', 'ARRAY', 'MAP', 'SET', 'RESET', 'SESSION', 'DATA', 'IF', IDENTIFIER, DIGIT_IDENTIFIER, QUOTED_IDENTIFIER, BACKQUOTED_IDENTIFIER}
ksql>
```

`CREATE STREAM` with quote

```
ksql> CREATE STREAM demo_as_json ("@timestamp" varchar, message varchar, node varchar) with (kafka_topic='demo', value_format='json');

 Message
----------------
 Stream created
----------------

ksql> describe demo_as_json;

Name                 : DEMO_AS_JSON
 Field      | Type
----------------------------------------
 ROWTIME    | BIGINT           (system)
 ROWKEY     | VARCHAR(STRING)  (system)
 @timestamp | VARCHAR(STRING)
 MESSAGE    | VARCHAR(STRING)
 NODE       | VARCHAR(STRING)
----------------------------------------
For runtime statistics and query details run: DESCRIBE EXTENDED <Stream,Table>;
```

Whilst this works, querying the column doesn't work: 

```
ksql> SET 'auto.offset.reset'='earliest';
Successfully changed local property 'auto.offset.reset' to 'earliest'. Use the UNSET command to revert your change.

ksql> select * from demo_as_json;
Code generation failed for SelectValueMapper
Caused by: Line 1, Column 15: Unexpected token "@"

ksql> select "@timestamp", message, node from demo_as_json;
Code generation failed for SelectValueMapper
Caused by: Line 1, Column 15: Unexpected token "@"

ksql> select "\@timestamp", message, node from demo_as_json;
Code generation failed for SelectValueMapper
Caused by: Line 1, Column 15: Unexpected token "@"

ksql> select \@timestamp, message, node from demo_as_json;
line 1:8: extraneous input '@' expecting {'(', 'NOT', 'NO', 'NULL', 'TRUE', 'FALSE', 'INTEGER', 'DATE', 'TIME', 'TIMESTAMP', 'INTERVAL', 'YEAR', 'MONTH', 'DAY', 'HOUR', 'MINUTE', 'SECOND', 'ZONE', 'CASE', 'PARTITION', 'STRUCT', 'REPLACE', 'EXPLAIN', 'ANALYZE', 'FORMAT', 'TYPE', 'TEXT', 'CAST', 'SHOW', 'TABLES', 'SCHEMAS', 'COLUMNS', 'COLUMN', 'PARTITIONS', 'FUNCTIONS', 'FUNCTION', 'ARRAY', 'MAP', 'SET', 'RESET', 'SESSION', 'DATA', 'IF', '+', '-', '*', STRING, BINARY_LITERAL, INTEGER_VALUE, DECIMAL_VALUE, IDENTIFIER, DIGIT_IDENTIFIER, QUOTED_IDENTIFIER, BACKQUOTED_IDENTIFIER}
```

**Comment by taylorpaul** [comment]
I think perhaps this issue should be closed? I was running into the issue described below and google brought me here, documenting my findings to help others:

I found in the [docs](https://docs.ksqldb.io/en/0.26.0-ksqldb/reference/sql/syntax/lexical-structure/#identifiers) that special characters should be supported by using backticks. I would expect this code to work, but I get the following error: `The TIMESTAMP column set in the WITH clause does not exist in the schema: 'TIMESTAMP'`

```
CREATE STREAM windows_event_stream (
    `@timestamp` STRING,
    message VARCHAR
) WITH (
    KAFKA_TOPIC='winlogbeat.raw', 
    KEY_FORMAT='KAFKA',
    PARTITIONS=3,
    TIMESTAMP='@timestamp',
    TIMESTAMP_FORMAT='yyyy-MM-dd''T''HH:mm:ss.SSSX', 
    VALUE_FORMAT='JSON'
);
```
Changed the code to also include backticks in the `WITH( )` portion fixed the issue. NOTE: I had to put the backticks in quotes for this to work! No error and a stream query worked:

```
CREATE STREAM windows_event_stream (
    `@timestamp` STRING,
    message VARCHAR
) WITH (
    KAFKA_TOPIC='winlogbeat.raw', 
    KEY_FORMAT='KAFKA',
    PARTITIONS=3,
    TIMESTAMP='`@timestamp`',
    TIMESTAMP_FORMAT='yyyy-MM-dd''T''HH:mm:ss.SSSX', 
    VALUE_FORMAT='JSON'
);
```
For ksqldb version I'm using CLI v0.26.0, Server v7.1.0 in CFK.
